src/rdallocator.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * librdkafka - Apache Kafka C library
+ *
+ * Copyright (c) 2019, Magnus Edenhill
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "rd.h"
+
+static void *
+rd_default_malloc(RD_UNUSED void *ud, size_t nsize) {
+	void *p = malloc(nsize);
+	rd_assert(p);
+	return p;
+}
+
+static void *
+rd_default_calloc(RD_UNUSED void *ud, size_t n, size_t sz) {
+	void *p = calloc(n, sz);
+	rd_assert(p);
+	return p;
+}
+
+static void *
+rd_default_realloc(RD_UNUSED void *ud, void *ptr, size_t nsize) {
+	void *p = realloc(ptr, nsize);
+	rd_assert(p);
+	return p;
+}
+
+static void
+rd_default_free(RD_UNUSED void *ud, void *ptr) {
+	free(ptr);
+}
+
+struct rd_allocator_state RD_CURRENT_ALLOCATOR = {
+	rd_default_malloc,
+	rd_default_calloc,
+	rd_default_realloc,
+	rd_default_free,
+	NULL
+};
+
+void rd_set_allocator(void *malloc, void *calloc, void *realloc, void *free, void *user_data) {
```

Comment:
keep below 80 columns.
rename arguments to avoid shadowing: `f_malloc`, `f_calloc`, etc.

Reply:
add an assert, like `rd_assert(!rd_allocator.is_set)` to make sure it is only set once, let this value be 0 for the default allocator.